ouroboros-network-framework/src/Ouroboros/Network/InboundGovernor.hs

```diff
@@ -329,14 +235,18 @@ inboundGovernor tracer serverControlChannel protocolIdleTimeout
             Just err -> traceWith tracer (TrMuxErrored connId err)
 
           -- the connection manager does should realise this on itself.
-          newState <- updateState (unregisterConnection connId state) connId
-          inboundGovernorLoop newState
+          let state' = unregisterConnection connId state
+          traceWith tracer ( TrInboundGovernorCounters
```

Comment:
I'm not sure if it is worth to trace before the recursive call because that is the first thing the recursive call is going to do!

Reply:
Good point, I'll remove it from all places but the first one.